cloudferrylib/utils/remote_runner.py

```diff
@@ -53,10 +53,17 @@ def run(self, cmd):
             with forward_agent(self.key):
                 LOG.debug("running '%s' on '%s' host as user '%s'",
                           cmd, self.host, self.user)
-                if self.sudo and self.user != 'root':
-                    return sudo(cmd)
-                else:
-                    return run(cmd)
+                res = None
+                #import pdb;pdb.set_trace()
+                try:
+                    if self.sudo and self.user != 'root':
+                        res = sudo(cmd)
+                    else:
+                        res = run(cmd)
+                except RemoteExecutionError as e:
```

Comment:
Exception must not be handled here, it must be handled by calling code. Please remove.